gnat2goto/driver/tree_walk.adb

```diff
@@ -2727,8 +2779,25 @@ package body Tree_Walk is
       --  Intern (Unique_Name (Corresponding_Spec (N)));
 
       Proc_Body   : constant Irep := Do_Subprogram_Or_Block (N);
-      Proc_Symbol : Symbol := Global_Symbol_Table (Proc_Name);
+      Proc_Symbol : Symbol;
    begin
+      if not Global_Symbol_Table.Contains (Proc_Name) then
```

Comment:
Should probably print a warning for missing procedures. You can write to stderr with `Put_Line(Standard_Error, "text");` not sure how to get a string out of a `Proc_Name` though.

Reply:
I think this would be hit if we encounter a subprogram for which we have not seen a declaration which would be legal ADA I think.

Reply:
Not sure how exactly this works. Ada, like C++, requires that declarations go before usage to resolve overloads and the like. Of course it counts stuffs that's in .ads files that are `with`-ed (or `new package`'ed) and I'm not sure if those end up in the global symbol table though (there should be _some_ way the frontend lets us access these though, otherwise we couldn't figure out what procedure we're actually calling in the backend).

Reply:
Yeah, currently we don't get an AST for the `.ads` file so we don't see any declarations that are in there. We should definitely improve this at some point to also process the `.ads` AST.

I think the case where we process a subprogram for which there is no symbol table entry yet might also happen when a function/procedure has been defined without a prior separate declaration.